### PR TITLE
SI-7933 REPL javax.script eval is cached result

### PR DESCRIPTION
**Pull request by rjolly**

### CODE REVIEW

File: src/repl/scala/tools/nsc/interpreter/IMain.scala [hunk 1]
```diff
@@ -1014,10 +1001,27 @@ class IMain(@BeanProperty val factory: ScriptEngineFactory, initialSettings: Set
...
+  def compile(reader: java.io.Reader): CompiledScript = compile(reader: String)
+
+  @throws[ScriptException]
+  def eval(script: String, context: ScriptContext): Object = compiled(script).eval(context)
 
   @throws[ScriptException]
-  def eval(reader: java.io.Reader, context: ScriptContext): Object = compile(reader).eval(context)
+  def eval(reader: java.io.Reader, context: ScriptContext): Object = eval(reader: String, context)
+
+  implicit def reader2string(reader: java.io.Reader) = {
```
Comment (adriaanm) [inline src/repl/scala/tools/nsc/interpreter/IMain.scala]: Maybe we could move this next to scala.tools.nsc.util.stringFromWriter?

**Comment by adriaanm** [comment]
please update the commit message with a short description of what the problem was and how it's solved

**Comment by adriaanm** [comment]
Thanks for clarifying, but I still have some questions.
Is this something that affects the repl under normal operation?

If so, I'm not sure we want to take that hit. Is there a way to only do this wrapping in CompiledScript when running in ScriptEngine mode?

**Comment by rjolly** [comment]
Adriaan,
It is already the case. The repl is not impacted at all by the change.

**Comment by adriaanm** [comment]
Ok, sorry, I'm a bit new to the repl code base.

**Pull request merged**